tools/standalone/Standalone.hs

```diff
@@ -0,0 +1,373 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TemplateHaskell #-}
+ {-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeApplications #-}
+
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+-- | Module: Standalone
+-- Copyright: Copyright © 2019 Kadena LLC.
+-- License: MIT
+-- Maintainer: Emmanuel Denloye-Ito <emmanuel@kadena.io>
+-- Stability: experimental
+--
+-- TODO
+--
+module Standalone where
+
+import Configuration.Utils hiding (disabled, Error)
+
+import Control.Concurrent
+import Control.Concurrent.Async
+import Control.DeepSeq
+import Control.Lens hiding ((.=))
+import Control.Monad
+import Control.Monad.Managed
+
+import Data.CAS.RocksDB
+import Data.LogMessage
+import Data.PQueue
+import Data.Typeable
+import qualified Data.TaskMap as TM
+import qualified Data.Text as T
+
+import GHC.Stats
+import GHC.Generics hiding (to)
+
+import qualified Network.HTTP.Client as HTTP
+import qualified Network.HTTP.Client.TLS as HTTPS
+
+import Numeric.Natural
+
+import PkgInfo
+
+import qualified Streaming.Prelude as S
+
+import System.Directory
+import qualified System.Logger as L
+import System.LogLevel
+import System.Timeout
+
+import Utils.Logging
+import Utils.Logging.Config
+import Utils.Logging.Trace
+
+-- chainweb imports
+
+import Chainweb.Chainweb
+import Chainweb.Chainweb.CutResources
+import Chainweb.Counter
+import Chainweb.Cut.CutHashes
+import Chainweb.CutDB
+import Chainweb.Graph
+import Chainweb.Logger
+import Chainweb.Logging.Amberdata
+import Chainweb.Logging.Config
+import Chainweb.Logging.Miner
+import Chainweb.Mempool.InMemTypes (MempoolStats(..))
+import Chainweb.Payload.PayloadStore
+import Chainweb.Sync.WebBlockHeaderStore.Types
+import Chainweb.Time
+import Chainweb.Utils
+import Chainweb.Version
+
+import Standalone.Chainweb
+import Standalone.Utils
+
+-- -------------------------------------------------------------------------- --
+-- Monitors
```

Comment:
I assume that all the monitors code is unchanged from the original code in `node/ChainwebNode.hs`. If that actually the case, we should consider moving that code in to a submodule in `node/` and use it for both binaries.

That can be done in a follow up PR.

Reply:
I'll keep this in mind.